src/lib/staged_ledger/staged_ledger.ml

```diff
@@ -2260,39 +2264,6 @@ module T = struct
             ( { Staged_ledger_diff.With_valid_signatures_and_proofs.diff }
             , invalid_on_this_ledger ) ) )
 
-  let latest_block_accounts_created t ~previous_block_state_hash =
```

Comment:
From what I can tell this is the biggest gain in this PR -- deleting this function and caching the `accounts_created` in the breadcrumb, you dont have to fold over the whole scan state every time you want this data